### PR TITLE
Add live-updating to form

### PR DESCRIPTION
**Pull request by stevenpetryk**
* Makes sidebar live-update as the user types and navigates the form
* Makes buttons live-update as the user types and navigates the form
* Makes it so buttons are enabled/disabled instead of disappearing
* Adds a proper UI state for loading ("Saving..." ... "Saved.")

### CODE REVIEW

File: app/assets/javascripts/registrations.js.erb [hunk 1]
```diff
@@ -7,17 +7,38 @@
...
       method: "PATCH",
       body: formData,
       credentials: 'same-origin',
       headers: new Headers({
-        "Accept": "application/json"
+        "Accept": "application/javascript"
       })
     })
+      .then(response => response.text())
+      .then(content => eval(content))
```
Comment (stevenpetryk) [inline app/assets/javascripts/registrations.js.erb]: I know `eval` is scary, but check out [this part of the diff](https://github.com/KnightHacks/knighthacks-web/pull/23/files#diff-f5cbf1be9e53e1d33a5fdf49ce4d8293), where I use the `j()` helper method to ensure the rendered HTML is properly and securely escaped.

File: app/workers/remind_user_to_finish_applying_worker.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ def perform(registration_id)
 
     Registration.transaction do
       registration.update(reminded_to_finish_applying: true)
-      RegistrationMailer.remind_registration_to_finish_applying_email(registration).deliver_now
+      RegistrationMailer.remind_user_to_finish_applying_email(registration).deliver_now
```
Comment (stevenpetryk) [inline app/workers/remind_user_to_finish_applying_worker.rb]: This just kinda snuck in here.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -23,7 +23,7 @@
     <footer role="main">
         <div class="container">
           <div class="links">
-            <%= link_to image_tag("logo-bw.png", class: "logo"), "/" %>
+            <%= link_to image_tag("logo-bw.svg", class: "logo"), "/" %>
```
Comment (stevenpetryk) [inline app/views/layouts/application.html.erb]: Doesn't belong in this PR but ¯\_(ツ)_/¯

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -21,9 +21,14 @@ def edit
...
+    @registration_form = RegistrationForm.new(current_registration, params[:section])
+
+    respond_to :js
+
     if current_registration.update(registration_params)
-      render json: current_registration
+      render :update
     else
+      # TODO
       render json: current_registration.errors.details, status: :unprocessable_entity
```
Comment (stevenpetryk) [inline app/controllers/registrations_controller.rb]: This is where the error handling will come in. Next PR (probably).

File: Gemfile [hunk 5]
```diff
@@ -15,8 +15,9 @@ gem "sidekiq"
 gem "simple_form"
 gem "uglifier", ">= 1.3.0"
 
-gem "momentjs-rails"
 gem "fetch-rails"
+gem "lodash-rails"
```
Comment (stevenpetryk) [inline Gemfile]: Just needed lodash for debounce. Might optimize this later if the JS bundle gets prohibitively big.

**Comment by maxcell** [comment]
This is really fucking awesome!!!!! If any of the form validation is too challenging, I would recommend that you just change it to be a drop down. Sometimes it is just easier.

**Comment by stevenpetryk** [comment]
True! I'll look into that when I start the validation stuff.
